src/main/java/org/mifos/connector/bulk/camel/routes/FileProcessingRoute.java

```diff
@@ -0,0 +1,110 @@
+package org.mifos.connector.bulk.camel.routes;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.databind.SequenceWriter;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.mifos.connector.bulk.schema.Transaction;
+import org.mifos.connector.bulk.schema.TransactionResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mifos.connector.bulk.zeebe.ZeebeVariables.FILE_NAME;
+
+@Component
+public class FileProcessingRoute extends BaseRouteBuilder {
+
+    private static final String TRANSACTION_LIST = "transactionList";
+    private static final String TRANSACTION_LIST_LENGTH = "transactionListLength";
+    private static final String TOTAL_AMOUNT = "totalAmount";
+    private static final String ONGOING_AMOUNT = "ongoingAmount";
+    private static final String FAILED_AMOUNT = "failedAmount";
+    private static final String COMPLETED_AMOUNT = "completedAmount";
+    private static final String LOCAL_FILE_PATH = "localFilePath";
+    private static final String RESULT_TRANSACTION_LIST = "resultTransactionList";
+    private static final String OVERRIDE_HEADER = "overrideHeader";
+
+    @Autowired
+    private CsvMapper csvMapper;
+
+    @Override
+    public void configure() throws Exception {
+
+        /**
+         * Parse the [Transaction] array from the csv file
+         * exchangeInput: [LOCAL_FILE_PATH] the absolute path to the csv file
+         * exchangeOutput: [TRANSACTION_LIST] containing the list of [Transaction]
+         */
+        from("direct:get-transaction-array")
+                .id("direct:get-transaction-array")
+                .log("Starting route direct:get-transaction-array")
+                .process(exchange -> {
+                    Double totalAmount = 0.0;
+                    Long failedAmount = 0L;
+                    Long completedAmount = 0L;
+                    String filename = exchange.getProperty(FILE_NAME, String.class);
+                    CsvSchema schema = CsvSchema.emptySchema().withHeader();
+                    FileReader reader = new FileReader(filename);
+                    MappingIterator<Transaction> readValues = csvMapper.readerWithSchemaFor(Transaction.class).with(schema).readValues(reader);
+                    List<Transaction> transactionList = new ArrayList<>();
+                    while (readValues.hasNext()) {
+                        Transaction current = readValues.next();
+                        transactionList.add(current);
```

Comment:
Since we are already breaking down a large batch file into subbatches in bulk processor, I feel there is no need to break this file any further. 

While addressing the review comments, I was breaking it down further. But after talking to Avik, I am going to apply the changes to process the file in a streamed manner. Do you have any reference code for streaming from S3 that you can point me to?